src/wasm/wasm-io.cpp

```diff
@@ -78,6 +88,20 @@ void ModuleReader::read(std::string filename, Module& wasm,
   }
 }
 
+void ModuleReader::readStdin(Module& wasm, std::string sourceMapFilename) {
+  std::vector<char> input = read_stdin(debug ? Flags::Debug : Flags::Release);
```

Comment:
Please add a TODO comment here that we are adding overhead here with an extra copy, and can perhaps optimize things better.